db/models/transactionentry.go

```diff
@@ -14,9 +23,11 @@ type TransactionEntry struct {
 	ParentID        int64             `bun:",nullzero"`
 	Parent          *TransactionEntry `bun:"rel:belongs-to"`
 	CreditAccountID int64             `bun:",notnull"`
+	FeeReserve      *TransactionEntry `bun:"rel:belongs-to"`
```

Comment:
It's added there in https://github.com/getAlby/lndhub.go/blob/main/lib/service/checkpayments.go#L68.
The `FeeReserveID` is not saved on the transaction entry, I could not figure out how to get the id of the inserted record in https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoices.go#L329 , I think because we're using a db transaction there it's not possible.

We're looking it up with its invoice id and entry_type = `fee_reserve` , and then adding it to its parent.

Reply:
so you say this `feeReserveEntry` here does not have an Id: https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoices.go#L329C34-L329C49

which then can be used in here: https://github.com/getAlby/lndhub.go/blob/main/lib/service/invoices.go#L311-L316 (if moved down)

by SQL it should be possible
and for documentation and proper references I would find it nice to have it, then the other query is also easier because we simply get the association.

Reply:
if bun does not do that automatically then I think we can use `returning` 
https://bun.uptrace.dev/guide/query-insert.html#api

with that we at least should have the inserted ID in the result: https://pkg.go.dev/database/sql#Result
which we then can use to set the foreign key. 

so we save the fee reserve, get the ID of that entry and save that in the tx and all this in a transaction. 
PG can do that.